substrate/frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1354,6 +1357,37 @@ pub mod pallet {
 	pub struct Pallet<T>(_);
 }
 
+/// This wrapper is created for handling the synchronization of [`Snapshot`], [`SnapshotMetadata`]
+/// and [`DesiredTargets`] storage items.
+pub struct SnapshotWrapper<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> SnapshotWrapper<T> {
+	/// Kill all snapshot related storage items at the same time.
+	pub fn kill() {
+		<Snapshot<T>>::kill();
+		<SnapshotMetadata<T>>::kill();
+		<DesiredTargets<T>>::kill();
+	}
+	/// Set all snapshot related storage items at the same time.
+	pub fn set(metadata: SolutionOrSnapshotSize, desired_targets: u32, buffer: Vec<u8>) {
+		<SnapshotMetadata<T>>::put(metadata);
+		<DesiredTargets<T>>::put(desired_targets);
+		sp_io::storage::set(&<Snapshot<T>>::hashed_key(), &buffer);
+	}
+
+	/// Check if all of the storage items exist at the same time or all of the storage items do not
+	/// exist.
+	#[cfg(feature = "try-runtime")]
+	pub fn is_consistent() -> bool {
+		(<Snapshot<T>>::exists() &&
+			<SnapshotMetadata<T>>::exists() &&
+			<DesiredTargets<T>>::exists()) ||
+			(!<Snapshot<T>>::exists() &&
+				!<SnapshotMetadata<T>>::exists() &&
+				!<DesiredTargets<T>>::exists())
+	}
```

Comment:
maybe can do something like this for improved readability
```suggestion
    pub fn is_consistent() -> bool {
        let snapshots = [
            <Snapshot<T>>::exists(),
            <SnapshotMetadata<T>>::exists(),
            <DesiredTargets<T>>::exists(),
        ];
        
        // All should either exist or not exist
        snapshots.iter().all(|&exists| exists) || snapshots.iter().all(|&exists| !exists)
}
```

Reply:
`snapshots.iter().all(|v| snapshots[0] == v)` has the same effect.